crates/core/component/dex/src/swap/proof.rs

```diff
@@ -133,8 +137,8 @@ impl ConstraintSynthesizer<Fq> for SwapCircuit {
         let balance_2 = BalanceVar::from_negative_value_var(swap_plaintext_var.delta_2_value());
         let balance_1_commit = balance_1.commit(transparent_blinding_var.clone())?;
         let balance_2_commit = balance_2.commit(transparent_blinding_var)?;
-        let transparent_balance_commitment = balance_1_commit + balance_2_commit;
-        let total_balance_commitment = transparent_balance_commitment + fee_commitment;
+        let opaque_commitment = fee_balance.commit(balance_blinding_var)?;
+        let total_balance_commitment = balance_1_commit + balance_2_commit + opaque_commitment;
```

Comment:
I'm not totally sure this solve the shared randomizer problem, and as I understand it the ECC recommendation is to 1/ use a balance-level blinding factor $\tilde v_i$ and to build a completely new generator $G_{\tilde v_i}$ *in addition* to $cv_f$ so that you cannot easily isolate either of the swap inputs e.g. $cv - cv_f = [-v_1]G_1 + [\tilde v_i]G_{\tilde v_i}$. But AFAICT, what we're doing here is to sample a random scalar $\tilde v_f$ and then never use it? So there's still a single randomizer for the entire balance commitment! Maybe I'm off but I would like to understand why!

Reply:
@TalDerei pointed out to me offline that the whole point of this approach is to yeet $cv_f$ out of the total balance commitment